.gitbook/3-protocol-devs/1-dep-specs/4-data-validation.md

```diff
@@ -0,0 +1,201 @@
+}
+```
+
+| Key                                   | Type   | Description                                                                  |
+| ------------------------------------- | ------ | ---------------------------------------------------------------------------- |
+| unsigned_certificate                  | Object | Unsigned certificate containing data validation information                  |
+| unsigned_certificate.cid              | string | A content identifier of a file in IPFS                                       |
+| unsigned_certificate.unique_id        | string | UniqueID of the oracle that validated the data                               |
+| unsigned_certificate.oracle_address   | string | Account address of the oracle that validated the data                        |
+| unsigned_certificate.deal_id          | int    | Deal to whom the provider intends to provide data                            |
+| unsigned_certificate.provider_address | string | Data provider's account address                                              |
+| unsigned_certificate.data_hash        | string | A hexadecimal string of a SHA256 hash value of the original data             |
+| signature                             | string | Base64-encoded string signed `unsigned_certificate` with Oracle private key. |
+
+
+### Data validation process
+
+#### Data Decryption
+Provider's encrypted data(`encrypted_data_base64`) can only be decrypted by oracle.
+
+```
+secret_key = SHA256(ECDH(oracle_private_key, provider_public_key))
+
+encrypted_data = Base64.Decode(encrypted_data_base64)
+
+orginal_data = AES256GCM.Decrypt(secret_key, encrypted_data)
+```
+
+#### Data Validation
+Verify that the original data matches the `data_hash`.
+```
+compare(data_hash, HEX.Encode(SHA256(orginal_data))
+```
+
+Verify that the `provider_address` of the original data matches the JWT auth token issuer of the request header.
+```
+compare(original_data.provider_address,jwtToken.issuer)
+```
+
+The deal information can be retrieved from Panacea using the deal ID
+
+Before verifying the data, it is checked whether the deal status is valid.
+If the Deal's status is invalid, oracle does not perform verification work.
+```
+compare(deal.status, 'DEAL_STATUS_ACTIVE')
+```
+
+Validate original data with `data_schema` extracted from Deal. 
+We currently support JSON schema.
+```
+data_schema = deal.data_schema
+
+JSONSchema.Validate(data_schema, orginal_data)
+```
+
+#### Data re-encryption and delivery via IPFS
+
+If data validation is successful, the data must be re-encrypted and stored on IPFS to be delivered to the consumer.
+
+To re-encrypt the data, a symmetric secret key must be used. The symmetric secret key can be derived by the following logic.
+```
+deal_id_bz = convertUint64ToBigEndian(deal_id)
+combined_key = SHA256(append(oracle_private_key, deal_id_bz, data_hash))
```

Comment:
It seems that the `secret key` and the `combined key` are mixed. How about unifying with a `secret key`?